terraform/ec2/main.tf

```diff
@@ -405,7 +405,7 @@ resource "null_resource" "setup_sample_app_and_mock_server" {
   provisioner "remote-exec" {
     inline = [
       "sudo yum update -y",
-      "sudo amazon-linux-extras install docker -y",
+      "sudo yum install -y docker",
```

Comment:
will this work for any AMI?

Reply:
The sample app and mock server is only launched on AL2023 instance. Previously it was on AL2022.

Reply:
Are these issues only present for the sample and mocked server host? Why not just use AL2 for sample app and mocked server then?

Reply:
We only care about testing the collector on AL3.

Reply:
True but AL2 will reach End of Life on 2025-06-30 so we have to upgrade then to AL3. So wanted to tackle 2 birds with 1 stone.